htdocs/comm/card.php

```diff
@@ -887,7 +887,8 @@
 						}
 					}
 					$relativepath = dol_sanitizeFileName($objp->ref).'/'.dol_sanitizeFileName($objp->ref).'.pdf';
-					print $formfile->showPreview($file_list, $propal_static->element, $relativepath, 0);
+					if (file_exists($conf->propal->multidir_output[$objp->entity].'/'.$relativepath))
```

Comment:
Because wec' are in a loop, doing a filexist generate a lot of io.
Do you know in which you case you have the file that does not exists ?

Reply:
It can happen when the file was deleted by a user for any reason, or the file was not generated.
I don't see another way to test if the file is existing

Reply:
I have modified the PR to not use file_exists()